pkg/services/preferences/preferences.go

```diff
@@ -0,0 +1,40 @@
+package preferences
+
+import (
+	"context"
+
+	"github.com/grafana/grafana/pkg/models"
+	pstore "github.com/grafana/grafana/pkg/services/preferences/store"
+	"github.com/grafana/grafana/pkg/services/sqlstore"
+	"github.com/grafana/grafana/pkg/setting"
+)
+
+type Service interface {
+	GetPreferencesWithDefaults(ctx context.Context, query *models.GetPreferencesWithDefaultsQuery) error
+	GetPreferences(ctx context.Context, query *models.GetPreferencesQuery) error
+	SavePreferences(ctx context.Context, query *models.SavePreferencesCommand) error
+}
+
+type Store struct {
+	PreferenceStore pstore.Store
+}
+
+func ProvideService(cfg *setting.Cfg, sqlstore sqlstore.Store) Service {
```

Comment:
We usually return the struct pointer here instead the interface. Is the idea to return the interface?

Reply:
yes, we want to have interfaces everywhere